### ISSUE TITLE
Define what is valid for a rotation scan

### ISSUE THREAD

**Issue by dperl-dls** [DiamondLightSource/python-artemis#376]
- can rotation be zero or less than zero?
- do we ever want to rotate another axis than omega?
- does omega have to stay within limits or can it rotate forever?
- what is the largest rotation we can or want to allow?

**Comment by dperl-dls** [comment]
@npat113 @ndg63276 @DominicOram would love your input on these or related questions

**Comment by DominicOram** [comment]
I'm not sure on the other points but for `do we ever want to rotate another axis than omega?` I would suggest that we either write it in a completely generic way or we write it mindful that at some point we probably will want to make it generic

**Comment by npat113** [comment]
rotation can be negative ( i.e. 180 to 0). In fact this is something we may explore in future.

Yes (not generally on here, but elsewhere this is valid)

Omega can run forever on I03

Max rotation should be set by beamline staff, not software devs. Current implementation in GDA is a mess, claims values not allowed but collects fine. Combination of omega speed, rotation range and maximum number of images - diamond tends to fall over when we tick from 5 to 6 numbers of frames.

**Comment by graeme-winter** [comment]
> * can rotation be zero or less than zero?

Yes, *but* I would want the NeXus file to invert the rotation axis and write +ve rotation values

> * do we ever want to rotate another axis than omega?

Yes: i19-1 do phi scans

> * does omega have to stay within limits or can it rotate forever?

Forever

> * what is the largest rotation we can or want to allow?

We have 1,000 frames / file and 6 digits for filename starting from 1, so that limits you to 999,999,000 frames. Since that would take 23 days to measure at 500 Hz and we only run 6 days / week I think we can call it "effectively imlimited"

**Comment by dperl-dls** [comment]
Noted, thanks

**Comment by DominicOram** [comment]
Will need to think about max velocities, these should be able to be read from the PV

**Comment by npat113** [comment]
also acceleration, time to reach constant speed for rotation scan or translations for line scan. Bluesky might already handle this, but not sure.

**Comment by dperl-dls** [comment]
Thanks for the input everyone, closing this now but feel free to add comments/reopen if something else comes to mind.
